### PR TITLE
[EKA-7] Encapsulate the patient object inside a response object for patient discovery. 

### PR DESCRIPTION
**Pull request by chethandeshpande**
Fixed the package names
Changed the contract to have patient object in discovery response instead of returning patient directly.
Moved request and response objects to respective folders.

### CODE REVIEW

File: health-information-provider-library/Patient/models/AuthenticationType.cs [hunk 1]
```diff
@@ -1,7 +1,8 @@
-namespace hip_library.Patient.models.dto
+namespace HipLibrary.Patient.Models
 {
     public enum AuthenticationType
     {
-        Direct, Mediated
```
Comment (ganesanarun) [inline health-information-provider-library/Patient/models/AuthenticationType.cs:5]: old files also getting pushed, I guess.

File: health-information-provider-library/Patient/ILink.cs [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
-using hip_library.Patient.models.dto;
+using HipLibrary.Patient.Models.Request;
+using HipLibrary.Patient.Models.Response;
 
-namespace hip_library.Patient
+namespace HipLibrary.Patient
 {
     public interface ILink
     {
         Task<Tuple<PatientLinkReferenceResponse, Error>> LinkPatients(PatientLinkReferenceRequest request);
```
Comment (ganesanarun) [inline health-information-provider-library/Patient/ILink.cs:10]:
```suggestion
        Task<Tuple<PatientLinkReferenceResponse, Error>> LinkPatients(PatientLinkReferenceRequest request);
        
```

File: health-information-provider-library/Patient/Models/Gender.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+namespace HipLibrary.Patient.Models
+{
+    public enum Gender
+    {
+        Male,
```
Comment (ganesanarun) [inline health-information-provider-library/Patient/Models/Gender.cs]: From HDCM, we are supposed to sent M/F/O.

File: health-information-provider-library/Patient/Models/IdentifierType.cs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+namespace HipLibrary.Patient.Models
+{
+    public enum IdentifierType
+    {
+        Mr,
```
Comment (ganesanarun) [inline health-information-provider-library/Patient/Models/IdentifierType.cs]:
```suggestion
        MR,
```

**Review by ganesanarun — CHANGES_REQUESTED** [review]

File: health-information-provider-library/Patient/Models/IdentifierType.cs (hunk 4 above)
Comment (chethandeshpande) [reply health-information-provider-library/Patient/Models/IdentifierType.cs]: According to naming conventions of Enum, it should be named in PascalCase. Shouldn't we be using that in our contract as well?

**Review by ganesanarun — APPROVED** [review]

**Pull request merged**